### PR TITLE
#354: Add namespace to the NewKubectlOptions constructor

### PR DESCRIPTION
**Pull request by j4ng5y**
Update the NewKubectlOptions constructor to require the namespace struct attribute as per #354

### CODE REVIEW

**Comment by j4ng5y** [comment]
So, in order to maintain backwards compatibility, I changed the NewKubectlOptions function to accept a variable length of strings and if we see that the number is 3 now, rather than original 2 case, then we set the namespace as well.

File: modules/k8s/kubectl_options.go [hunk 1]
```diff
@@ -2,18 +2,28 @@ package k8s
...
-		ContextName: contextName,
-		ConfigPath:  configPath,
+// NewKubectlOptions will return a pointer to new instance of KubectlOptions with the configured options
+func NewKubectlOptions(opts ...string) *KubectlOptions {
+	switch {
+	case len(opts) = 3:
+		return &KubectlOptions{
+			ContextName: opts[0],
+			ConfigPath:  opts[1],
+			Namespace:   opts[2],
```
Comment (yorinasub17) [inline modules/k8s/kubectl_options.go]: While you are in here, can you also initialize the map for `Env`? Basically add `Env: map[string]string{}` to the struct creation. The reason is that by default it initializes to `nil`, and you can't start adding to `Env`, so by initializing, it makes the experience better.

Comment (j4ng5y) [reply modules/k8s/kubectl_options.go]: Will do

Comment (j4ng5y) [reply modules/k8s/kubectl_options.go]: added this in 12eaf30

Comment (yorinasub17) [reply modules/k8s/kubectl_options.go]: Thanks!

**Comment by yorinasub17** [comment]
Thanks! Kicking off the build now.

**Comment by j4ng5y** [comment]
The helm stuff is probably going to fail, I forgot to update those. I'm adding the fixes now.

**Comment by yorinasub17** [comment]
Ah it looks like it also failed on `goimports` in `modules/k8s/kubectl_options.go`, probably due to the misalignment in the struct initialization for `Env`. Can you run `gofmt` (or `goimports`) on that file?

**Comment by yorinasub17** [comment]
Thanks! Kicked off another build.

**Comment by yorinasub17** [comment]
Got a few more build failures resulting from:
```
# github.com/gruntwork-io/terratest/modules/k8s
../k8s/client.go:22:30: not enough arguments in call to NewKubectlOptions
        have (string, string)
        want (string, string, string)
```

**Comment by j4ng5y** [comment]
Ok, in places that don't use the options, I'm going to use the "default" namespace, which seems to happen all over the place so that will take me just a minute

**Comment by yorinasub17** [comment]
Ok kicking off another test build!

**Comment by j4ng5y** [comment]
I'm having a fairly hard time figuring out why that one build ... looked like an AMI thing

**Comment by yorinasub17** [comment]
The AMI failure is flaky and happens from time to time due to the number of packer builds we run in parallel. However, the `setup-minikube` failure is new to me and it's weird because I know this was working this morning in a different repo. I am investigating that now...

**Comment by yorinasub17** [comment]
Ah it looks like this branch is quite a bit behind master and does not include the fixes for minikube (https://github.com/gruntwork-io/terratest/commit/cf0c4c521bc299d6814455dfe0a81d056ae39aa5). Can you rebase on master?

**Comment by j4ng5y** [comment]
re-basing now

**Comment by yorinasub17** [comment]
Thanks! Kicking off a new build.

**Comment by j4ng5y** [comment]
S3 bucket failure this time

**Comment by yorinasub17** [comment]
Looks like there are also a few more build failures that were not caught before, due to the build tags:

```
modules/helm/install_test.go:31:41: not enough arguments in call to k8s.NewKubectlOptions
	have (string, string)
	want (string, string, string)
modules/helm/upgrade_test.go:32:41: not enough arguments in call to k8s.NewKubectlOptions
	have (string, string)
	want (string, string, string)
modules/k8s/service_account_test.go:45:44: not enough arguments in call to NewKubectlOptions
	have (string, string)
	want (string, string, string)
```

**Comment by j4ng5y** [comment]
Cool, i'll clean those up this morning.

File: modules/k8s/config.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func CopyHomeKubeConfigToTemp(t *testing.T) string {
 	return path
 }
 
-// CopyHomeKubeConfigToTemp will copy the kubeconfig in the home directory to a temp file.
+// CopyHomeKubeConfigToTempE will copy the kubeconfig in the home directory to a temp file.
```
Comment (yorinasub17) [inline modules/k8s/config.go]: Good catch! Thanks!

File: modules/k8s/errors.go [hunk 3]
```diff
@@ -51,10 +54,12 @@ type PodNotAvailable struct {
 	pod *corev1.Pod
 }
 
+// Error is a simple function to return a formatted error message as a string
 func (err PodNotAvailable) Error() string {
 	return fmt.Sprintf("Pod %s is not available", err.pod.Name)
 }
 
+// NewPodNotAvailableError returnes a PodNodAvailable struct when Kubernetes deems a pod is not available
```
Comment (yorinasub17) [inline modules/k8s/errors.go]:
```suggestion
// NewPodNotAvailableError returnes a PodNotAvailable struct when Kubernetes deems a pod is not available
```

Comment (j4ng5y) [reply modules/k8s/errors.go]: Spelling is hard :smile:

**Comment by j4ng5y** [comment]
I fixed the go-imports issue if you wouldn't mind building again

**Comment by j4ng5y** [comment]
I miss another call to NewKubectlOptions, but I fixed that. Also, a few S3 tests failed. Not sure whats up with that :smile:

**Comment by j4ng5y** [comment]
Horray! Everything built :)

**Comment by yorinasub17** [comment]
Ok I just reviewed the change set again and this looks good to me! I am going to go ahead and merge this in. Thanks for your patience and responsiveness working with me through the issues!

**Comment by yorinasub17** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.20.0

**Pull request merged**